src/basins/edgetracking.jl

```diff
@@ -0,0 +1,194 @@
+            distance = dist(current_state(pds, 1) - current_state(pds, 2))
+            T += dt
+            if output_level>1
+                push!(edge, (current_state(pds, 1) + current_state(pds, 2))/2)
+            end
+        end
+        u1, u2 = bisect_to_edge(pds, mapper; eps=eps1)
+        edgestate = (u1 + u2)/2
+        correction = dist(edgestate - state)
+        counter += 1
+
+        if output_level>0
+            push!(bisect_pts1, u1)
+            push!(bisect_pts2, u2)
+            push!(edge, edgestate)
+        end
+        
+        verbose && println("... Iteration $(counter): Edge at $(edgestate)")
+        (counter == maxiter) && @warn("Reached maximum number of $(maxiter) iterations;
+        did not converge.")
+    end
+
+    (counter < maxiter) && println("Edge-tracking converged after $(counter) iterations.")
+
+    (output_level == 0) && return edgestate
+    (output_level == 1) && return edge
+    (output_level == 2) && return [edge, bisect_pts1, bisect_pts2]
+end;
+
+"""
+    bisect_to_edge(pds::ParallelDynamicalSystem, mapper::AttractorMapper; eps=1e-9)
+Finds the basin boundary between two states `u1, u2 = current_states(pds)` by bisecting along a
+straight line in phase space. The states `u1` and `u2` must belong to different basins.
+Returns two new states located on either side of the basin boundary at a maximum 
+(Euclidean) distance of `eps` between each other.
+
+## Keyword arguments
+* `eps = 1e-9`: The maximum (Euclidean) distance between the two returned states.
+
+## Description
+`pds` is a `ParallelDynamicalSystem` with two states. The `mapper` must be an `AttractorMapper`
+of subtype `AttractorsViaProximity` or `AttractorsViaRecurrences`.
+
+Note: If the straight line between `u1` and `u2` intersects the basin boundary multiple
+times, the method will find one of these intersection points. If more than two attractors
+exist, one of the two returned states may belong to a different basin than the initial
+conditions `u1` and `u2`. A warning is raised if the bisection involves a third basin.
+"""
+function bisect_to_edge(pds::ParallelDynamicalSystem, mapper::AttractorMapper; eps=1e-9)
+    u1, u2 = current_states(pds)
+    idx1, idx2 = mapper(u1), mapper(u2)
+    
+    if idx1 == idx2
+        error("Both initial conditions belong to the same basin of attraction.")
+    end
+    
+    distance = dist(u1-u2)
+    while distance > eps
+        u_new = (u1 + u2)/2
+        idx_new = mapper(u_new)
+        # slightly shift u_new if it lands directly on the boundary
+        if idx_new == -1
+            u_new += eps*(u_1 - u_2)
+        end
+        # update u1 or u2
+        if idx_new == idx1
+            u1 = u_new
+        else 
+            u2 = u_new
+            if idx_new != idx2
+                @warn "New bisection point belongs to a third basin of attraction."
+            end
+        end    
+        distance = dist(u1-u2)
+    end
+    [u1, u2]
+end;
+
+dist(x) = sqrt(sum(abs, x.^2));
```

Comment:
Your `dist` is literally `LinearAlgebra.norm` so you most defnitely don't have to define this :D

Reply:
But as I explained above, in fact, you won't be calling `LinearAlgebra.norm`, as you will define `diffnorm`.

Reply:
Yes, I have defined `diffnorm`.